### ISSUE TITLE
Argument aliasing check

### ISSUE THREAD

**Issue by tmmsartor** [giaf/blasfeo#30]
As pointed out in #27 aliasing of routines arguments can lead to unexpected and unwanted behaviors.
To mitigate this problem I propose two incremental solutions:
- Specify this requirement more clearly in the documentation 
- Add a compilation flag with which one can enable some kind of checks for argument aliasing on all routines that do not allow them.

**Comment by giaf** [comment]
Yep this was exactly the idea of the run_time_checks, but so far it has been implemented only in very few routines.
